system/shruntime.py

```diff
@@ -441,22 +441,22 @@ def run_pipe_sequence(self, pipe_sequence,
             # outside of the actual command execution, i.e. exec_py_file
             # exec_sh_file, e.g. command not found, not executable etc.
             except ShFileNotFound as e:
-                err_msg = '%s\n' % e.message
+                err_msg = '%s\n' % e.args[0]
                 if self.debug:
                     self.logger.debug(err_msg)
                 self.stash.write_message(err_msg)
                 # set exit code to 127
                 current_state.return_value = 127
                 break  # break out of the pipe_sequence, but NOT pipe_sequence list
             except Exception as e:
-                err_msg = '%s\n' % e.message
+                err_msg = '%s\n' % e.args[0]
```

Comment:
`err_msg = '%s\n' % e`

When catching `Exception`, we have no idea if the exception has a useful message (or any message at all), so it's best to include the exception type.